photon-serde/message_data_types.yaml

```diff
@@ -5,29 +5,35 @@ bool:
   java_type: bool
   cpp_type: bool
   java_decode_method: decodeBoolean
+  java_encode_shim: encodeBoolean
 int16:
   len: 2
   java_type: short
   cpp_type: int16_t
   java_decode_method: decodeShort
   java_list_decode_method: decodeShortList
+  java_encode_shim: encodeShort
```

Comment:
Should this be python encode_shim? Same for the rest of the file changes

Reply:
I think a couple of things are a bit funky with this generation stuff as it is now. I didn't want to risk breaking things that were working just to get my stuff in.

There is precedent for me doing it that way. This code snippet is pulled from line 75 of the same file for python things.

```
{%- elif field.type | is_intrinsic %}
        # {{ field.name }} is of intrinsic type {{ field.type }}
        ret.{{field.name}} = packet.{{ type_map[field.type].java_decode_method }}()
```

The thing that's a particular smell to me is this snippet from `photon-serde/messages.yaml`. I think the way I have it for encode is correct because it just names the shim but the decode one is annoying because it makes a call to packet as well.

```yaml
  python_decode_shim: packet.decodeTransform
  python_encode_shim: encodeTransform
```

I can change this to do more clean-up, but I would rather do that later when I can add more tests and get automatic reporting when things are cooked.

Reply:
Yeah that's sus for sure. Probably just an oversight on my part, Python support was thrown together quickly lol

Reply:
Don't sweat it.

Is this something you guys want fixed now or can we do it as a follow up to Sim things when there's more active tests? I think any changes to generated things should be subject to as much testing as possible

Reply:
If running the message serde code only changes Python code, and you're satisfied that it's been tested thoroughly enough, go for it in this PR.

Reply:
I would personally rather do it in a subsequent pr. The Sim one is getting harder to do dev on because it's dependent on doing all these things going in. I don't want to keep rebasing with merge conflicts when I can just make changes with my testing as a part of the next one.